src/Sulu/Bundle/CoreBundle/DependencyInjection/SuluCoreExtension.php

```diff
@@ -225,4 +227,80 @@ private function initCache($cache, ContainerBuilder $container, Loader\XmlFileLo
 
         $loader->load('cache.xml');
     }
+
+    /**
+     * Initializes list builder.
+     *
+     * @param ContainerBuilder $container
+     * @param Loader\XmlFileLoader $loader
+     */
+    private function initListBuilder(ContainerBuilder $container, Loader\XmlFileLoader $loader)
+    {
+        $loader->load('list_builder.xml');
+
+        $metadataPaths = $this->getBundleMappingPaths($container->getParameter('kernel.bundles'), 'list-builder');
+        $fileLocator = $container->getDefinition('sulu_core.list_builder.metadata.file_locator');
+        $fileLocator->replaceArgument(0, $metadataPaths);
+
+        $generalMetadataCacheFolder = $this->createOrGetFolder('%sulu.cache_dir%/list-builder/general', $container);
+        $doctrineMetadataCacheFolder = $this->createOrGetFolder('%sulu.cache_dir%/list-builder/doctrine', $container);
+
+        $container->setParameter('sulu_core.list_builder.metadata.provider.general.cache_dir', $generalMetadataCacheFolder);
+        $container->setParameter('sulu_core.list_builder.metadata.provider.doctrine.cache_dir', $doctrineMetadataCacheFolder);
+    }
+
+    /**
+     * Create and return directory.
+     *
+     * @param string $directory
+     * @param ContainerBuilder $container
+     *
+     * @return string
+     */
+    protected function createOrGetFolder($directory, ContainerBuilder $container)
+    {
+        $directory = $container->getParameterBag()->resolveValue($directory);
+        if (!file_exists($directory)) {
```

Comment:
We can use the `Symfony\Filesystem\Filesystem` class here:

```
if (false === $filesystem->exists($directory)) {
    $filesystem->mkdir($directory);
}
```

Also -- note that `777` is a security issue - anybody on the server could edit the cache files and run any PHP code they want as the webserver user.